### DISCUSSION TITLE
Does removing an extension preserves its settings if I reinstall it later?

### DISCUSSION THREAD

**Discussion by lamyergeier** [FreshRSS/FreshRSS#6213] [Q&A]
I am using docker version of Freshrss. To update an extension I have  to remove the extension first by doing

```
docker exec freshrss rm -rf "/var/www/FreshRSS/extensions/${Extension}"
```

and then add it again as follows (after doing git pull)

```
docker cp "${Extension}" freshrss:/var/www/FreshRSS/extensions
```

But will this preserve the extension settings after removing it? Is there a better way to update an extension?

**Comment by Alkarex** [comment]
I believe it does, but the best would be to just test

**Reply by lamyergeier** [reply]
So is this the way to update an extension with docker? Or there is a better way?

**Reply by Alkarex** [reply]
I am not sure what you mean, but the suggested way with Docker is to have a Docker volume for your extensions, so that they are persisted when updating FreshRSS https://github.com/FreshRSS/FreshRSS/blob/edge/Docker/README.md
Extensions can then be updated e.g. by git in their own folders
